### ISSUE TITLE
require-hook flags forEach loops

### ISSUE THREAD

**Issue by Ariane-B** [jest-community/eslint-plugin-jest#1046]
Sometimes, we have looped tests, where we want to test a lot of assertions with a similar logic.

We define a constant with an array of test conditions and results, then we run a `forEach` loop on them. The rule says my `forEach` should be in a hook, which is weird.

Example of looped tests:

```javascript
      const GET_ENTITY_BY_INDEX_TESTS = deepFreeze([
         {
            textAskedFor: `the first draft section`,

            entityTypeForTest: ENTITY_TYPES.sections,
            entityIdForTest: SECTION_DRAFT_1_ID,
            formStatusForTest: FORM_STATUSES.draft,
            indexForTest: 0,
            parentIdForTest: DRAFT_FORM_ID,

            valueExpected: ENTITY_GOTTEN_ARTIFICIALLY,
            shouldThrow: false,
         },
         {
            textAskedFor: `the second draft section`,

            entityTypeForTest: ENTITY_TYPES.sections,
            entityIdForTest: SECTION_DRAFT_2_ID,
            formStatusForTest: FORM_STATUSES.draft,
            indexForTest: 1,
            parentIdForTest: DRAFT_FORM_ID,

            valueExpected: ENTITY_GOTTEN_ARTIFICIALLY,
            shouldThrow: false,
         },
         {
            textAskedFor: `the first draft control measure`,

            entityTypeForTest: ENTITY_TYPES.controlMeasures,
            entityIdForTest: CONTROL_MEASURE_DRAFT_1_ID,
            formStatusForTest: FORM_STATUSES.draft,
            indexForTest: 0,
            parentIdForTest: SECTION_DRAFT_1_ID,

            valueExpected: ENTITY_GOTTEN_ARTIFICIALLY,
            shouldThrow: false,
         },
         {
            textAskedFor: `the first non-draft control measure`,

            entityTypeForTest: ENTITY_TYPES.controlMeasures,
            entityIdForTest: CONTROL_MEASURE_NON_DRAFT_1_ID,
            formStatusForTest: FORM_STATUSES.nonDraft,
            indexForTest: 0,
            parentIdForTest: SECTION_NON_DRAFT_1_ID,

            valueExpected: ENTITY_GOTTEN_ARTIFICIALLY,
            shouldThrow: false,
         },
         {
            textAskedFor: `a control measure with an impossibly high index`,

            entityTypeForTest: ENTITY_TYPES.controlMeasures,
            entityIdForTest: undefined,
            formStatusForTest: FORM_STATUSES.draft,
            indexForTest: INVALID_HIGH_INDEX,
            parentIdForTest: SECTION_DRAFT_1_ID,

            valueExpected: undefined,
            shouldThrow: true,
         },
         {
            textAskedFor: `a control measure with an impossibly low index`,

            entityTypeForTest: ENTITY_TYPES.controlMeasures,
            entityIdForTest: undefined,
            formStatusForTest: FORM_STATUSES.draft,
            indexForTest: INVALID_LOW_INDEX,
            parentIdForTest: SECTION_DRAFT_1_ID,

            valueExpected: undefined,
            shouldThrow: true,
         },
         {
            textAskedFor: `a control measure whose parent doesn't exist.`,

            entityTypeForTest: ENTITY_TYPES.controlMeasures,
            entityIdForTest: CONTROL_MEASURE_DRAFT_1_ID,
            formStatusForTest: FORM_STATUSES.draft,
            indexForTest: 0,
            parentIdForTest: INVALID_ID,

            valueExpected: undefined,
            shouldThrow: true,
         },
      ])

      GET_ENTITY_BY_INDEX_TESTS.forEach((testInfo) => {
         let returnValueText

         if (testInfo.shouldThrow) {
            returnValueText = `throw an error`
         } else if (testInfo.valueExpected === ENTITY_GOTTEN_ARTIFICIALLY) {
            returnValueText = `return entity with ID ${ testInfo.entityIdForTest } and entity type ${ testInfo.entityTypeForTest }`
         } else if (typeof testInfo.valueExpected == 'undefined') {
            returnValueText = `return undefined`
         } else {
            returnValueText = `return ${ JSON.stringify(testInfo.valueExpected) }`
         }

         it(`[Looped] getEntityByIndex SHOULD ${ returnValueText } IF asked for ${ testInfo.textAskedFor }.`, async () => {
            const parentEntityTypeForTest = ENTITY_TYPE_RELATIONSHIPS[testInfo.entityTypeForTest].parent

            const getEntityByIdMock = jest.fn(
               () => jest.fn()
               // First call: Returns parent
                  .mockReturnValueOnce(NORMALIZED_DATA_DEFAULT[testInfo.formStatusForTest].entities[parentEntityTypeForTest][testInfo.parentIdForTest])
               // Second call: Returns the entity we're looking for
                  .mockReturnValue(NORMALIZED_DATA_DEFAULT[testInfo.formStatusForTest].entities[testInfo.entityTypeForTest][testInfo.entityIdForTest]),
            )

            const store = await createStore({ storeOverrides: { getters: { getEntityById: getEntityByIdMock } } })

            function getEntity() {
               return store.getters['criticalRiskForms/getEntityByIndex']({
                  parentId: testInfo.parentIdForTest,
                  entityType: testInfo.entityTypeForTest,
                  index: testInfo.indexForTest,
                  formStatus: testInfo.formStatusForTest,
               })
            }

            let entityReturned

            if (testInfo.shouldThrow) {
               expect(() => {
                  entityReturned = getEntity()
               }).toThrow(expect.anything())
            } else {
               entityReturned = getEntity()

               let finalValueExpected

               if (testInfo.valueExpected === ENTITY_GOTTEN_ARTIFICIALLY) {
                  finalValueExpected = store.getters['criticalRiskForms/dataByReadonlyOrFormStatus']({ formStatus: testInfo.formStatusForTest })
                     .entities[testInfo.entityTypeForTest][testInfo.entityIdForTest]
               } else {
                  finalValueExpected = testInfo.valueExpected
               }

               expect(entityReturned).toStrictEqual(finalValueExpected)
            }
         })
      })
```

**Comment by G-Rath** [comment]
What you're doing is exactly what [`it.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) is for - is there any reason why you're not using that?

Otherwise I'd recommend using an inline disable directive.

**Comment by G-Rath** [comment]
Closing per #1047
